### PR TITLE
Update old code

### PR DESCRIPTION
**Pull request by Mubelotix**
I copied a few data structures from the meilisearch core project when I started this SDK.
It seems that these structs and enums were modified during the last months and that breaking changes were introduced.
It would be useful if GitHub allowed me to subscribe to the changes on the 2 concerned files but it's not possible.

### TODO

- [x] Update the outdated code
- [x] Add new tests so that we can detect breaking changes faster (note that test structure is not ideal and there will be a new PR about this)

### CODE REVIEW

**Comment by curquiza** [comment]
Hello @Mubelotix!

>  It would be useful if GitHub allowed me to subscribe to the changes on the 2 concerned files but it's not possible.

Can you give me the names of the file you would watch? I can try to take a look at them on each release!

Is this PR over? Can you convert it to a draft PR if it's not finished? 🙂 Once you've done it you can open it (= make it green again) and request me as reviewer! I will request someone in the core-team myself if needed only.

**Comment by Mubelotix** [comment]
> Can you give me the names of the file you would watch? I can try to take a look at them on each release!
I can try to take a look at these files myself but they are MeiliSearch/meilisearch-core/src/update/mod.rs and MeiliSearch/meilisearch-core/src/settings.rs for information.

> Is this PR over? Can you convert it to a draft PR if it's not finished? slightly_smiling_face Once you've done it you can open it (= make it green again) and request me as reviewer! I will request someone in the core-team myself if needed only.
It wasn't but now it is! I forgot to select "draft PR".

> Can you describe what kind of current problem we have? I will open an issue about this!
I am going to open the issue myself it will be faster.

**Comment by curquiza** [comment]
> I can try to take a look at these files myself but they are MeiliSearch/meilisearch-core/src/update/mod.rs and MeiliSearch/meilisearch-core/src/settings.rs for information.

Got it! That's also my job to make sure this package is up-to-date 😉 

> It wasn't but now it is! I forgot to select "draft PR".

You can also change it anytime by clicking here:

Capture d’écran 2020-12-10 à 11 07 02

Not really obvious, I know 😅 

> I am going to open the issue myself it will be faster. And we are going to fix this in another PR.

You're awesome!

I will review your PR soon!

File: src/progress.rs
Comment (curquiza) [inline src/progress.rs:79]: What is the usage of `SettingsUpdate`? The user is not able to update the `primary_key` from the HTTP API, so I'm not sure this line was useful.

**Review by curquiza — CHANGES_REQUESTED** [review]
One remark but all the changes seem good 👍

File: src/progress.rs
Comment (Mubelotix) [reply src/progress.rs:79]: I don't why it's available. This structure is sent by the server. I made everything available without thinking.

Comment (curquiza) [reply src/progress.rs:79]: Can you remove it then? I talk to the core-team and this structure does not represent what is the server sends, so this is not a concern of this SDK. There will never send a `primary_key` in the settings.

Comment (Mubelotix) [reply src/progress.rs:79]: Yeah, it seems you are right! I looked at the Meilisearch code and this seems to be dead code. A primary key cannot be updated with settings so it should not be possible for the server to notify the client that the key was updated. I will do a PR on meilisearch/meilisearch to fix this.

Comment (curquiza) [reply src/progress.rs:79]: Yes, it's useful for an internal purpose, this is not dead code. The core team told me you should avoid using the meilisearch-core structures since this is not what the MeiliSearch server will send. This is only internal.
It works for the moment, so we can let it as it is, but they do not recommend following exactly those files 🙂

**Comment by aidanhs** [comment]
Would love to see progress on this, given that applying settings with facets isn't possible with this SDK atm :)

**Comment by curquiza** [comment]
Waiting for the removal of the `primary_key` by @Mubelotix and we will merge it 🙂

**Comment by curquiza** [comment]
Hello @Mubelotix!
Thanks for the fix!
There are still merge conflicts you might fix with a rebase before the merge! 😇

**Comment by Mubelotix** [comment]
I applied your suggestion @curquiza. It worked but I had to force push which is not that great. Would not have been better to merge `main` into this branch instead of rebasing?

**Review by curquiza — APPROVED** [review]
Awesome!!! 🎉

**Comment by curquiza** [comment]
bors merge

**Pull request merged**
